app/ht_indexer/src/ht_queue_service/queue_producer.py

```diff
@@ -1,70 +1,124 @@
+    def publish_messages(self, queue_message: Dict[str, Any]) -> None:
 
+        # TODO: Define a dataclass for the queue message to ensure type safety and validation
+        # TODO: Define a retry mechanism with exponential backoff for publishing messages
+
+        """
+        Publish the message. If the channel is closed during publishing, it will attempt to reconnect and retry
+        publishing. The message is serialized to JSON format before publishing.
+        :param queue_message: The message to be published should be a dictionary.
+        :return: None
+
+        :raises: Exception if the message cannot be serialized or published.
+        :raises: pika.exceptions.ChannelClosed and pika.exceptions.ConnectionClosed if the channel is closed during publishing.
+        :raises: pika.exceptions.ConnectionClosed if the connection is closed during publishing.
+        :raises: pika.exceptions.StreamLostError if the stream is lost during publishing.
+        :raises: pika.exceptions.ChannelWrongStateError if the channel is in the wrong state during publishing.
+        :raises: Other exceptions that may occur during publishing.
+        """
+        message_id = get_queue_message_id(queue_message)
         try:
+            if not self.channel or self.channel.is_closed:
+                logger.warning(f"Channel closed before publish (queue: {self.queue_manager.queue_name}). Reconnecting.")# self.channel = self.channel_creator.get_channel()   # Reopen channel after exception
+                self.queue_reconnect()
+
+            try:
+                body = json.dumps(queue_message)
+            except (TypeError, ValueError) as json_err:
+                logger.error(
+                    f"Failed to serialize message {queue_message.get('ht_id')}: {json_err}", exc_info=True
+                )
+                raise
 
-            if not self.ht_channel or self.ht_channel.is_closed:
+            if not self.channel or self.channel.is_closed:
+                logger.warning("Channel is closed before publish. Reconnecting.")
                 self.queue_reconnect()
+                raise pika.exceptions.ChannelClosed("Channel was closed")
 
-            body = json.dumps(queue_message)
-            self.ht_channel.confirm_delivery() # Ensure the channel is in confirm mode
-            self.ht_channel.basic_publish(
-                exchange=self.exchange, routing_key=self.queue_name, body=body,
-                properties=pika.BasicProperties(delivery_mode=2, content_type="application/json") # make message persistent
-            )
-            logger.info(f"Published message to {self.queue_name}: {body}")
+            self.channel.basic_publish(
+                                exchange=self.queue_manager.main_exchange_name,
+                                routing_key=self.queue_manager.queue_name,
+                                body=body,
+                                properties=pika.BasicProperties(delivery_mode=2,
```

Comment:
Is there a constant defined anywhere in pika like `DELIVERY_MODE_PERSISTENT` or something that we can use instead of a literal `2` here?